### DISCUSSION TITLE
What is the Typescript type for .$transaction array?

### DISCUSSION THREAD

**Discussion by felinto-dev** [prisma/prisma#10096] [Q&A] [answered]
Hey!

A common case use for .$transaction is to get multiple items from a table + count, for example:

```ts
    const [items, count] = await this.prisma.$transaction([
      this.prisma.platformAccount.findMany(query),
      this.prisma.platformAccount.count({ where: query.where }),
    ]);

    return { items, count };
```

I want to create a helper function to help me do this without manually creating a .$transaction every time. I believe Prisma could improve at this point.

However, the VS Code doesn't give me any hints about which typing I should use.

![image](https://user-images.githubusercontent.com/32253743/140252516-e384140d-c277-4a59-aab3-6f2970219478.png)

![image](https://user-images.githubusercontent.com/32253743/140252561-6f66810e-9e0b-48be-9aa0-94b34a241eb4.png)

I don't get any VS Code suggestions!

What Typescript Type should I use?

**Comment by austincrim** [comment, marked as answer]
Hi @felinto-dev 👋, thanks for the question.

Your intuition was right in that you will need to use the `PrismaPromise<T>` type here. You can import it from the `@prisma/client` package.

Here's how I typed `prismaPromises`:
```ts
import { PrismaPromise, PlatformAccount } from '@prisma/client'

const prismaPromises: [PrismaPromise<PlatformAccount[]>, PrismaPromise<number>] = [
  this.prisma.platformAccount.findMany(query),
  this.prisma.platformAccount.count({where: query.where})
]

```

I used TypeScript's [tuple feature](https://www.typescriptlang.org/docs/handbook/basic-types.html#tuple) to denote `prismaPromises` will only ever have those two elements in that order. After doing this, the result of your call to `$transaction` should have the correct type inferred.

Let me know if this works for you!

**Reply by felinto-dev** [reply]
Thanks! <3

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
